src/openrct2-ui/drawing/engines/HardwareDisplayDrawingEngine.cpp

```diff
@@ -98,7 +101,40 @@ class HardwareDisplayDrawingEngine final : public X8DrawingEngine
             }
         }
 
-        _screenTexture = SDL_CreateTexture(_sdlRenderer, pixelFormat, SDL_TEXTUREACCESS_STREAMING, width, height);
+        sint32 scaleQuality = gConfigGeneral.scale_quality;
+        if (gConfigGeneral.use_nn_at_integer_scales &&
+            gConfigGeneral.window_scale == std::floor(gConfigGeneral.window_scale))
+        {
+            scaleQuality = 0;
+        }
+        if (scaleQuality == 3)
```

Comment:
This is getting a bit cryptic, please create an enum.

Reply:
Done